### ISSUE TITLE
loss not descent, while tensorflow do! Do I code in wrong way? Help!

### ISSUE THREAD

**Issue by deahhh** [pytorch/pytorch#63546]
## ❓ Questions and Help

I code a small model in pytorch 1.7.1+cu110, python 3.8.10, ubuntu 20.04, native, as follows:

def test_torch():
    import torch
    import torch.nn as nn
    from torch.utils import data
    from torch.utils.data import dataloader

    data = torch.randn(100000, 2)
    y = torch.sin(data[:, 0]*data[:, 1]).unsqueeze(1)
    print(y.shape)
    data = torch.cat([data, y], dim=1)

    model = nn.Sequential(
        nn.Linear(2, 100),
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 100), 
        nn.ReLU(),
        nn.Linear(100, 1)
    )

    for p in model.parameters():
        print(p.shape, p.requires_grad)

    criterion = torch.nn.L1Loss()
    opt = torch.optim.SGD(model.parameters(), 0.001)

    dl = dataloader.DataLoader(data, batch_size=500)
    for epoch in range(100):
        for d in dl:
            pred = model(d[:, :2])
            loss = criterion(pred, d[:, 2])
            opt.zero_grad()
            loss.backward()
            opt.step()
            print("epoch:[{}], loss: {}".format(epoch, loss.item()))
def test_tf():
    import tensorflow as tf
    import numpy as np
    from tensorflow import keras
    model = tf.keras.Sequential([
            keras.layers.Dense(units=10, activation='relu', input_shape=[2]), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=10, activation='relu'), 
            keras.layers.Dense(units=1), 
        ]
    )
    model.compile(optimizer=keras.optimizers.Adam(1e-3), loss="mean_squared_error")
    xs = np.random.randn(100000, 2).astype(np.float32)
    ys = np.sin(xs[:,0] * xs[:, 1])
    model.fit(xs, ys, epochs=100, batch_size=500)
if __name__ == "__main__":
    test_torch()

### Please note that this issue tracker is not a help form and this issue will be closed.
epoch:[99], loss: 0.405108243227005
epoch:[99], loss: 0.4149976968765259
epoch:[99], loss: 0.4175589084625244
epoch:[99], loss: 0.42489686608314514
epoch:[99], loss: 0.4205479323863983
epoch:[99], loss: 0.41913238167762756
epoch:[99], loss: 0.4025334417819977

loss never descent to 0.1 in test_torch(), while down to 1e-4 in test_tf().

**Comment by lukasfolle** [comment]
Please post your request here https://discuss.pytorch.org/.

**Comment by mruberry** [comment]
As @lukasfolle notes, this question is more appropriate for the PyTorch forums (at least to start).
